frontend/packages/kubevirt-plugin/src/k8s/patches/vm/vm-cdrom-patches.ts

```diff
@@ -0,0 +1,171 @@
+import { last, flatten, includes } from 'lodash';
+import {
+  BOOT_ORDER_SECOND,
+  BOOT_ORDER_FIRST,
+  PVC_ACCESSMODE_RWM,
+  PVC_VOLUMEMODE_FS,
+  getBootableDevicesInOrder,
+} from 'kubevirt-web-ui-components';
+import { getName } from '@console/shared';
+import { Volume, K8sResourceKind } from '@console/internal/module/k8s';
+import {
+  WINTOOLS_CONTAINER_NAMES,
+  StorageType,
+} from '../../../components/modals/cdrom-vm-modal/constants';
+import { CD } from '../../../types/vm';
+import { VMKind } from '../../../types';
+import {
+  getVolumes,
+  getDataVolumeTemplates,
+  getDisks,
+  getVolumeDataVolumeName,
+} from '../../../selectors/vm';
+
+const getNextAvailableBootOrderIndex = (vm: VMKind) => {
+  const sortedBootableDevices = getBootableDevicesInOrder(vm);
+  const numBootableDevices = sortedBootableDevices.length;
+  const lastBootableDevice: any = last(sortedBootableDevices);
+
+  // assigned indexes start at two as the first index is assigned directly by the user
+  return numBootableDevices > 0 ? lastBootableDevice.value.bootOrder + 1 : BOOT_ORDER_SECOND;
+};
+
+const assignBootOrderIndex = (vm: VMKind, currDevBootOrder = -1) => {
+  let bootOrder = currDevBootOrder;
+  if (currDevBootOrder !== BOOT_ORDER_FIRST) {
+    bootOrder = getNextAvailableBootOrderIndex(vm);
+  }
+  return bootOrder;
+};
+
+export const getCDsPatch = (
+  vm: VMKind,
+  cds: CD[],
+  pvcs: K8sResourceKind[],
+  storageClasses: K8sResourceKind[],
+) => {
+  let newBootOrder = assignBootOrderIndex(vm);
+  let DISKS = getDisks(vm);
+  let VOLS = getVolumes(vm);
+  let DATATEMPLATES = getDataVolumeTemplates(vm);
+
+  const patches = cds
+    .filter((cd) => cd.changed)
+    .map(
+      ({ name, pvc, type, bootOrder, bus, container, windowsTools, url, storageClass, size }) => {
+        const pvcObject = pvcs.find((vol) => getName(vol) === pvc) || pvcs[0];
```

Comment:
`pvcs[0]` might not be valid as the `pvcs` is passed as `persistentVolumesClaims.data` from the modal where  `persistentVolumesClaims` is optional in `CDRomModalProps`